core/src/main/java/io/aiven/klaw/helpers/db/rdbms/InsertDataJdbc.java

```diff
@@ -160,20 +163,27 @@ public synchronized Map<String, String> insertIntoRequestConnector(
   }
 
   public synchronized String insertIntoTopicSOT(List<Topic> topics) {
-    topics.forEach(
-        topic -> {
-          log.debug("insertIntoTopicSOT {}", topic.getTopicname());
-          if (!topic.isExistingTopic()) {
-            TopicID topicID = new TopicID();
-            topicID.setTenantId(topic.getTenantId());
-            topicID.setTopicid(topic.getTopicid());
-
-            if (topicRepo.existsById(topicID)) {
-              topic.setTopicid(getNextTopicRequestId("TOPIC_ID", topic.getTenantId()));
-            }
-          }
-          topicRepo.save(topic);
-        });
+    Set<Topic> existingTopicIds = new HashSet<>();
+    topicRepo
+        .findAllById(
+            topics.stream()
+                .filter(t -> !t.isExistingTopic())
+                .map(t -> new TopicID(t.getTopicid(), t.getTenantId()))
+                .collect(Collectors.toList()))
+        .forEach(existingTopicIds::add);
+
+    for (Topic topic : existingTopicIds) {
+      log.info("insertIntoTopicSOT {}", topic.getTopicname());
```

Comment:
Hey @snuyanzin I think we can switch these info statements to debug statments.

Reply:
good catch
seems I missed it 
thanks, fixed